Code refactoring

EnzoD86 opened the pull request.

lucasnz commented:
This integration is for a different brand of IR senders, but it could be worth aligning with it:
https://github.com/smartHomeHub/SmartIR/blob/master/docs/CLIMATE.md

They use `temperature_sensor` instead of `temp_sensor` in the config.

EnzoD86 commented:
Hi, I'm glad this repository is being used by someone! 
Your pull request is interesting; I had in mind to make these changes if this repository was actually used by others too. 
The only change I want to make to your pull request is to insert a parameter (for example "eu" or "us") to select the server (so as to avoid having to enter the entire URL of the API). I will post the changes shortly.
Thank you for your contribution!

EnzoD86 commented:
Hi, I merged your changes into the main branch by making some changes (as I mentioned before).
A new key has been added in the configuration called "country" and which allows you to choose which API the component must use.
I added the humidity sensor (so in your pull request) and I changed the name of the configuration key for the temperature sensor (as you recommended).
When you have time, update your configuration and let me know if it works correctly for the US country too.
Thanks for your help.

lucasnz commented:
I've done a basic test and your changes appear to work fine :)

EnzoD86 committed:
Code refactoring and improve logger

EnzoD86 committed:
Code refactoring
